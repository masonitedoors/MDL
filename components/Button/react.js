import React from 'react'
import classNames from 'classnames/bind'
import PropTypes from 'prop-types'
import { mCheck } from '@masonite/svg-icons'
import s from './style.module.scss'
import Icon from '../Icon/react'

const cx = classNames.bind(s)

const Button = ({
  children,
  className,
  disabled,
  fullWidth,
  onClick,
  variant,
  size,
  style,
  uppercase,
  active,
  ...props
}) => {
  const classes = cx('btn', {
    'btn--primary': variant === 'primary',
    'btn--secondary': ['secondary', 'dark'].includes(variant),
    'btn--light': variant === 'light',
    'btn--default': variant === 'default',
    'btn--uppercase': uppercase,
    'btn--sm': size === 'small' || size === 'sm',
    'btn--large': size === 'large',
    'btn--disabled': disabled,
    'btn--full-width': fullWidth,
    'btn--active': active,
    'btn--large-active': size === 'large' && active,
    'btn--small-active': (size === 'small' || size === 'sm') && active,
  }, className)

  return (
    <button
      type="button"
      className={classes}
      onClick={onClick}
      disabled={disabled}
      style={style}
      {...props}
    >
      {children}
      {active && size === 'large' && (
        <span className={cx('btn--active-check')}>
          <Icon svg={mCheck} size={16} />
        </span>
      )}
    </button>
  )
}

Button.propTypes = {
  active: PropTypes.bool,
  children: PropTypes.string.isRequired,
  className: PropTypes.string,
  onClick: PropTypes.func,
  disabled: PropTypes.bool,
  variant: PropTypes.oneOf(['default', 'primary', 'secondary', 'light']),
  size: PropTypes.oneOf(['small', 'sm', 'large']),
  style: PropTypes.shape({}),
  uppercase: PropTypes.bool,
  fullWidth: PropTypes.bool,
}

Button.defaultProps = {
  active: false,
  className: null,
  disabled: false,
  onClick: null,
  variant: 'default',
  size: null,
  style: null,
  uppercase: false,
  fullWidth: false,
}

export default Button
