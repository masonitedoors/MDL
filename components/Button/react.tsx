import React, { FC, ReactNode } from 'react'
import classNames from 'classnames/bind'
import { mCheck } from '@masonite/svg-icons'
import s from './style.module.scss'
import Icon from '../Icon/react'

const cx = classNames.bind(s)

export interface ButtonProps {
  children: ReactNode
  className?: string
  disabled?: boolean
  fullWidth?: boolean
  onClick?: any
  variant?: 'default' | 'primary' | 'secondary' | 'light'
  size?: 'sm' | 'lg' | 'small' | 'large'
  style?: any
  uppercase?: boolean
  active?: boolean
  bold?: boolean
}

const Button: FC<ButtonProps> = ({
  children,
  className,
  disabled,
  fullWidth,
  onClick,
  variant,
  size,
  style,
  uppercase,
  active,
  bold,
  ...props
}) => {
  const classes = cx(
    'btn',
    {
      'btn--primary': variant === 'primary',
      'btn--secondary': ['secondary', 'dark'].includes(variant),
      'btn--light': variant === 'light',
      'btn--default': variant === 'default',
      'btn--uppercase': uppercase,
      'btn--sm': size === 'small' || size === 'sm',
      'btn--large': size === 'large',
      'btn--disabled': disabled,
      'btn--full-width': fullWidth,
      'btn--bold': bold,
      'btn--active': active,
      'btn--large-active': size === 'large' && active,
      'btn--small-active': (size === 'small' || size === 'sm') && active,
    },
    className,
  )

  return (
    <button
      type="button"
      className={classes}
      onClick={onClick}
      disabled={disabled}
      style={style}
      {...props}
    >
      {children}
      {active && size === 'large' && (
        <span className={cx('btn--active-check')}>
          <Icon svg={mCheck} size={16} />
        </span>
      )}
    </button>
  )
}

export default Button
