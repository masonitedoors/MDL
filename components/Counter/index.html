<div class="Counter [Counter--hidden]">{{ value }}</div>
