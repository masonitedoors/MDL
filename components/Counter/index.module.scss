@import "styles/index";

.Counter {
  display: inline-block;
  font-size: 12px;
  color: $color-night;
  padding: 0 $space-xs;
  background: $color-silver;
  line-height: 1.4;
  border-radius: 100px;
}

.Counter--hidden {
  visibility: hidden;
}
