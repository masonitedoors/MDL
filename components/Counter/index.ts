export { default as React } from './react'
