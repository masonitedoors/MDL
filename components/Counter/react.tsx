import React, { FC, HTMLAttributes } from 'react'
import classNames from 'classnames/bind'
import styles from './index.module.scss'

const cx = classNames.bind(styles)

export interface CounterProps {
  /** Display the counter when it is equal to zero. */
  showZero?: boolean

  /** The displayed number */
  value: number
}

export const Counter: FC<HTMLAttributes<HTMLDivElement> & CounterProps> = ({
  className,
  showZero = false,
  value,
  ...props
}) => (
  <div
    className={cx('Counter', { 'Counter--hidden': !showZero && +value === 0 }, className)}
    {...props}
  >
    {value}
  </div>
)

export default Counter
