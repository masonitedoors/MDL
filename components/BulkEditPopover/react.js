import React, { useEffect } from 'react'
import classNames from 'classnames/bind'
import { React as Checkbox } from 'components/Checkbox'
import { React as TextField } from 'components/TextField'
import styles from './style.module.scss'

const cx = classNames.bind(styles)

const BulkEditPopover = React.forwardRef(
  (
    {
      formControlType,
      formControlProps,
      left = 0,
      top = 0,
      width = 0,
      popDirection = 'top',
      visible = false,
      focused = false,
      onMouseLeave,
      clearAllCellStates,
      value,
    },
    ref,
  ) => {
    const bulkEditOutsideClickHandler = event => {
      if (ref.current) !ref.current.contains(event.target) && clearAllCellStates()
    }

    useEffect(() => {
      window.addEventListener('click', bulkEditOutsideClickHandler, true)
      return window.removeEventListener('click', bulkEditOutsideClickHandler)
    }, [])

    const getDirectionStyle = () => {
      const style = {
        top: {
          bottom: '100%',
          top: 'auto',
        },
        left: {
          right: '100%',
          left: 'auto',
        },
      }
      return style[popDirection]
    }

    switch (formControlType) {
      case TextField:
        formControlProps = {
          removeMargin: true,
          ...formControlProps,
        }
        break
      default:
    }

    return (
      <div
        ref={ref}
        className={cx('bulk-edit-popover', {
          'bulk-edit-popover--visible': visible || focused,
          'bulk-edit-popover--focused': focused,
          'bulk-edit-popover--w-textfield': formControlType === TextField,
          'bulk-edit-popover--w-checkbox': formControlType === Checkbox,
        })}
        style={{
          top,
          left,
          width,
          ...getDirectionStyle(),
        }}
        onMouseLeave={onMouseLeave}
        data-testid="bulkEditPopover"
      >
        {formControlType && React.createElement(formControlType, { ...formControlProps })}
      </div>
    )
  },
)

export default BulkEditPopover
