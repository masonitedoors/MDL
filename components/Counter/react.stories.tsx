import React from "react";
import { Story, Meta } from "@storybook/react";
import Counter from "./react";

const Template = (args) => <Counter {...args} />;

export default {
  title: "Counter",
  component: Counter,
} as Meta;

export const Primary = Template.bind({});

Primary.args = {
  value: 42,
};
