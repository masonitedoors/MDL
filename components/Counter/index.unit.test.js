import React from 'react'
import { render } from '@testing-library/react'
import Counter from './index'

describe('Counter', () => {
  it('should render without crashing', () => {
    const { asFragment } = render(<Counter value={5} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
