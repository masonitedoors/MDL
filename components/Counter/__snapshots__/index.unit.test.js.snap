// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountPill should render without crashing 1`] = `
<DocumentFragment>
  <div
    class="CountPill"
  />
</DocumentFragment>
`;
